src/middlewared/middlewared/plugins/pool.py

```diff
@@ -692,6 +695,132 @@ def disk_to_stripe():
                     if os.path.exists(os.path.join("/dev", name)):
                         yield name
 
+    @item_method
+    @accepts(Int('id'), Dict(
+        'options',
+        Str('label', required=True),
+        Str('disk', required=True),
+        Bool('force', default=False),
+        Str('passphrase', password=True),
+    ))
+    @job(lock='pool_replace')
+    async def replace(self, job, oid, options):
+        """
+        Replace a disk on a pool.
+
+        `label` is the ZFS guid or a device name
+        `disk` is the identifier of a disk
+        """
+        pool = await self._get_instance(oid)
+
+        verrors = ValidationErrors()
+
+        unused_disks = await self.middleware.call('disk.get_unused')
+        disk = list(filter(lambda x: x['identifier'] == options['disk'], unused_disks))
+        if not disk:
+            verrors.add('options.disk', 'Disk not found.', errno.ENOENT)
+        else:
+            disk = disk[0]
+
+        if not options['force'] and not await self.middleware.call(
+            'disk.check_clean', disk['devname']
```

Comment:
`disk` might be `None` here, probably we should have raised `verrors` above

Reply:
True, fixing!